ci/single_binary_test.d

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env rdmd
+module system_integration_test;
+
+private:
+
+import std.algorithm;
+import std.file;
+import std.format;
+import std.path;
+import std.process;
+import std.stdio;
+import core.thread;
+
+/// Root of the repository
+immutable RootPath = __FILE_FULL_PATH__.dirName.dirName;
+immutable IntegrationPath = RootPath.buildPath("tests").buildPath("system");
+immutable LauncherPath = RootPath.buildPath("tests").buildPath("launcher");
+
+/+ ***************************** Commands to run **************************** +/
+/// A simple test to ensure that the container works correctly,
+/// e.g. all dependencies are installed and the binary isn't corrupt.
+immutable BuildBinary = [ "dub", "build", "-c", "launcher", ];
+immutable BinaryUp = [ "build/agora-launcher", "-c", LauncherPath.buildPath("configs.yaml"), ];
+immutable BinaryDown = [ "killall", "-9", "agora-launcher", ];
+immutable RunIntegrationTests = [ "dub", "--root", IntegrationPath, ];
+immutable Cleanup = [ "rm", "-rf", LauncherPath.buildPath("node/0/.cache/"),
+                      LauncherPath.buildPath("node/1/.cache/"),
+                      LauncherPath.buildPath("node/2/.cache/")];
+
+private int main (string[] args)
+{
+    runCmd(BinaryDown, false);
+
+    runCmd(BuildBinary);
+
+    // First make sure that there we start from a clean slate,
+    // as the docker-compose bind volumes
+    runCmd(Cleanup);
+    Thread.sleep(1.seconds);
```

Comment:
You shouldn't need to `sleep`, if wait is called correctly.

Reply:
I removed `sleep`.